validator/sawtooth_validator/journal/chain.py

```diff
@@ -383,7 +383,7 @@ def _find_common_height(self, new_chain, cur_chain):
             # walk the current chain back until we find the block that is the
             # same height as the new chain.
             while (cur_blkw.block_num > new_blkw.block_num
-                   and new_blkw.previous_block_id != NULL_BLOCK_IDENTIFIER):
+                   and cur_blkw.previous_block_id != NULL_BLOCK_IDENTIFIER):
```

Comment:
Could you please also correct the pre-existing cut and paste comments at lines? remove ~362 and correct ~366.